### PR TITLE
CVE-2023-44487

### PR DESCRIPTION
**Pull request by spolti**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: agent.Dockerfile [hunk 1]
```diff
@@ -5,11 +5,11 @@ FROM golang:1.20 as builder
 WORKDIR /go/src/github.com/kserve/kserve
 COPY go.mod  go.mod
 COPY go.sum  go.sum
-
 RUN go mod download
 
 COPY pkg/    pkg/
 COPY cmd/    cmd/
+RUN go mod tidy
```
Comment (israel-hdez) [inline agent.Dockerfile]: May you, please, remove changes to this file?
The `go mod tidy` is more a dev command that should not be called here when doing the build.

Comment (spolti) [reply agent.Dockerfile]: oh yeah, it is a left over that I forgot to remove.
thanks

Comment (spolti) [inline agent.Dockerfile]:
```suggestion
```

**Pull request merged**